### ISSUE TITLE
Compatibility with React 17

### ISSUE THREAD

**Issue by Teyras** [transcriptic/react-map-interaction#39]
`componentWillReceiveProps` will be deprecated in React 17. Do you plan to address this?

**Comment by zachcowell** [comment]
This warning is firing for me when debugging

```
backend.js:1 Warning: componentWillReceiveProps has been renamed, and is not recommended for use. See https://fb.me/react-async-component-lifecycle-hooks for details.

* Move data fetching code or side effects to componentDidUpdate.
* If you're updating state whenever props change, refactor your code to use memoization techniques or move it to static getDerivedStateFromProps. Learn more at: https://fb.me/react-derived-state
* Rename componentWillReceiveProps to UNSAFE_componentWillReceiveProps to suppress this warning in non-strict mode. In React 17.x, only the UNSAFE_ name will work. To rename all deprecated lifecycles to their new names, you can run `npx react-codemod rename-unsafe-lifecycles` in your project source folder.
```

**Comment by robertabenassi** [comment]
I am using this library in a small project that need to be React 17 compliant. I am using ComponentDidUpdate instead of componentWillReceiveProps. 
But I am really interested in this featue added soon.

**Comment by scottyantipa** [comment]
Fix is in the works

**Comment by scottyantipa** [comment]
I think the way to go here is to change this component to be either fully controlled, or uncontrolled.

like https://reactalt.org/blog/2018/06/07/you-probably-dont-need-derived-state.html

Then there isn't the dance being done in componentWillReceiveProps because we aren't trying to sync props and state. This would break callers though, so it would be a major update. The new API would require the `onChange` method to be used in the case where you want to have control over scale/translation. Or dont use onChange in which case it is fully uncontrolled.

**Comment by scottyantipa** [comment]
I proposed a fix in #57 if anyone wants to comment there

**Comment by scottyantipa** [comment]
@Teyras @zachcowell @robertabenassi Any thoughts on #57

**Comment by Teyras** [comment]
This looks like a step in the right direction to me. Thank you!

**Comment by robertabenassi** [comment]
@scottyantipa 
sorry for the late feedback. It seems ok for me now.
Thanks a lot!

**Comment by scottyantipa** [comment]
This is now released in 2.0.0. Thanks everyone.
